### PR TITLE
Add get_best_hyperparameters method

### PR DESCRIPTION
**Pull request by omalleyt12**

### CODE REVIEW

File: kerastuner/engine/base_tuner.py [hunk 1]
```diff
@@ -220,23 +220,46 @@ def on_search_end(self):
...
+
+        # Arguments:
+            num_trials: (int, optional). Number of `HyperParameters` objects to
+              return. `HyperParameters` will be returned in sorted order based on
+              trial performance.
+
+        # Returns:
+            List of `HyperParameter` objects.
+        """
+        return [t.hyperparameters for t in self.oracle.get_best_trials(num_trials)]
```
Comment (fchollet) [inline kerastuner/engine/base_tuner.py]: Why not just return the `values` dict? This is what people are going to be looking for most of the time. If looking for the actual hp object, they can get it easily from `get_best_trials`.

Comment (omalleyt12) [reply kerastuner/engine/base_tuner.py]: The `values` dict by itself doesn't have logic to handle conditional parameters, so if those were present it couldn't be used to recreate the best model. My thinking was that the main usecase for this method would be to pass its results to `build_model` to retrain the best model on the full dataset, and `build_model` expects the `HyperParameters` object

**Comment by omalleyt12** [comment]
Thanks for the review!

File: kerastuner/engine/base_tuner.py (hunk 1 above)
Comment (fchollet) [reply kerastuner/engine/base_tuner.py]: Ok, makes sense.

**Comment by omalleyt12** [comment]
Thanks for the review!

**Pull request merged**